tests/reference_images.rs

```diff
@@ -95,49 +97,135 @@ fn check_references() {
             let mut path: Vec<_> = path.components().collect();
             (path.pop().unwrap(), path.pop().unwrap())
         };
+
+        let mut filename_parts = filename
+            .as_os_str()
+            .to_str()
+            .unwrap()
+            .split('.')
+            .collect::<Vec<_>>();
+
+        // Ignore the file extension
+        filename_parts.pop().unwrap();
+
+        // The penultimate part of `filename_parts` represents the metadata.
+        let meta_str = filename_parts.pop().unwrap();
+        let mut meta = meta_str.split('_').collect::<Vec<_>>();
+        let ref_crc;
+        let anim_frame: Option<usize>;
+
+        if meta.len() == 1 {
+            // `CRC`
+            ref_crc = meta.pop().unwrap();
+            anim_frame = None;
+        } else if meta.len() == 3 && meta[0] == "anim" {
+            // `anim_FRAME_CRC`
+            ref_crc = meta.pop().unwrap();
+            let frame: usize = meta.pop().unwrap().parse().unwrap();
+            anim_frame = Some(frame.checked_sub(1).expect("frame number cannot be 0"));
+        } else {
+            panic!(
+                "unrecognized reference image metadata format: {:?}",
+                meta_str
+            );
+        }
+
+        // The remaining part represents the original file name
+        let orig_filename = filename_parts.join(".");
+
+        // Check the reference image's CRC
+        let ref_crc = u32::from_str_radix(ref_crc, 16).unwrap();
+
+        let ref_crc_actual = {
+            let mut hasher = Crc32::new();
+            hasher.update(&*ref_img);
+            hasher.finalize()
+        };
+
+        if ref_crc_actual != ref_crc {
+            panic!(
```

Comment:
This test, comparing the crc of the test suite with the actual crc of the test suite image is new, right? It doesn't hurt and will properly trigger for faulty new test suite images (good) but should also be an `unreachable!`.

But it seems to have *replaced* the test for crc of `test_img`. Please restore the crc test for the opened `test_img`.